backend/routes/referralRouters.ts

```diff
@@ -42,22 +45,64 @@ const checkUserIsManager = (
     next();
   } else {
     try {
-      throw new StatusCodedError("unauthroized request: not a manager", 401);
+      throw new StatusCodedError("forbidden request: not a manager", 403);
     } catch (err) {
       next(err);
     }
   }
 };
 
-referralRouter.get( //I feel like we should restrict this to only the specific job creator and no one else
-  "/jobs",
-  checkUserIsManager,
-  async (req: Request, res:Response, next: NextFunction) => {
-    try{
-      const jobId = req.query.jobId ? (req.query.jobId as string) : "";
-      const referrals = await referralControllers.getReferralsFromJobPostId(jobId);
-      res.status(200).json({referrals});
-    }catch(e:any){
+const checkIfQueryHasValidJobId = async (
+  req: Request,
+  res: Response,
+  next: NextFunction
+) => {
+  if (!req.query.jobId) {
+    next(new StatusCodedError("bad request: no jobId in query", 400));
+  } else if (
+    !(await jobPostControllers.getOneJobPostWithId(req.query.jobId as string))
+  ) {
+    next(
+      new StatusCodedError("bad request: no such jobFeed with input jobId", 400)
+    );
+  } else {
+    next();
+  }
+};
+
+const checkOneJobPostIsCreatedByUser = async (
+  req: Request,
+  res: Response,
+  next: NextFunction
+) => {
+  const jobPostId = req.query.jobId as string; //checkIfQueryHasValidJobId checks that it does exist.
+  const jobPostCreatorId = (
+    (await jobPostControllers.getOneJobPostWithId(jobPostId)) as JobPost
+  ).hiringManagerId; //checkIfQueryHasValidJobId checks that the jobPostId is the valid one.
+  if (req.user?.id === jobPostCreatorId) {
```

Comment:
I believe this line should say

if(req.user?.id !== jobPostCreatorId){ //logged in user does not match jobPostCreator so throw error
    next(new StatusCodedError("forbidden, not the jobpost creater", 403))
}

Reply:
It actually is. U can push the change